app/konnect/gateway-manager/plugins/update-custom-plugin.md

```diff
@@ -0,0 +1,142 @@
+
+When a schema is updated in {{site.konnect_short_name}}, the {{site.konnect_short_name}} 
+platform doesn't trigger payload reconciliation automatically.
+This means that if you don't make any configuration changes, such as adding, 
+modifying, or deleting a {{site.base_gateway}} entity, the data plane nodes won't receive a
+payload update.
+
+When pushing changes to the control plane, the payload reconciliation only affects 
+data plane nodes if an instance of the plugin that uses the updated schema has its configuration 
+changed.
+
+Since plugin configurations are stored as JSON blobs, a schema change alone doesn't impact the 
+plugin configuration. However, if the plugin itself is updated, the new schema affects how 
+the new plugin configuration is represented.
+
+### Custom plugin update path
+
+When you need to make plugin changes, we recommend updating the schema in 
+{{site.konnect_short_name}} first, and then on the data plane nodes:
+
+1. Start a migration window.
+1. Update plugin schema in {{site.konnect_short_name}}.  
+1. (Optional) Update configuration for existing plugin instances.
+
+    Based on the nature of the schema updates, you might need this optional step 
+    after updating the schema in {{site.konnect_short_name}} to make sure that any
+    existing plugin entities are updated before the schemas are 
+    changed in the data plane nodes.
+
+1. Update plugin schema on each data plane node.
+1. Stop the migration window.
+
+{:.important}
+> **Important**: In cases where a breaking change is made to the schema, we **do not** 
+recommend updating data plane nodes first. 
+The change will affect each data plane node as soon as the node receives its first payload. 
+This will happen even if there are no changes in the control plane for existing or new plugins 
+using the updated schema. 
+
+See the following table for a comparison of possible changes and upgrade paths, in the case of 
+a configuration parameter change in a custom plugin's schema:
+
+| | Required Default | Required Non-default | Non-required Default | Non-required Non-default |
+|--|--|--|--|--|
+| Add | ✅ | ⏸️ | ✅ | ✅ |
+| Remove | ⏸️ | ❌ | ⏸️ | ⏸️ |
+| Type Change | ❌ | ❌ | ❌ | ❌ |
+
+{:.note .no-icon}
+> **Legend:**
+* ✅ Happy path (1 → 2 → 4 → 5)
+* ⏸️ Semi-happy path (1 → 2 → 3 → 4 → 5)
+* ❌ Unhappy path (1 → 2 → DP Error → 4 → 5)
+> <br><br>
+> **Note:** Unhappy paths don’t break existing proxy functionality, but they do cause temporary 
+`Out of Sync` states until both the configured plugins and the data plane nodes are updated with 
+the new schemas.
+
+#### Adding or deleting fields
+
+If new fields in a schema don't have default values and aren't required, a payload update won't break data plane payload validation. This means that even if new plugins are added or existing ones are updated, the data plane remains in sync because null fields are ignored.
+
+Here's an example of a non-breaking change to a schema:
```

Comment:
Not sure if we need the example because I'm not sure which values are default or required because I'm unfamiliar with this plugin. Maybe you could add a link to the plugin doc if you'd like to keep the example so people know which values are default/required?

Reply:
You don't need to be familiar with the plugin, you just need to see that no defaults or required attributes are set in the example. I added some explanation to the example intro text to point out what to look at.